catalog/spatial/registry/registry-policy-plugin/src/main/java/org/codice/ddf/registry/policy/RegistryPolicyPlugin.java

```diff
@@ -157,42 +143,43 @@ private void parsePermissionsFromString(List<String> permStrings,
         }
     }
 
-    private PolicyResponse getWritePolicy(Metacard input, Map<String, Serializable> properties) {
-        HashMap<String, Set<String>> policy = new HashMap<>();
-        if (Requests.isLocal(properties) && input != null
-                && (RegistryUtility.isInternalRegistryMetacard(input)
-                || RegistryUtility.isRegistryMetacard(input))) {
+    private PolicyResponse getWritePolicy(Metacard input, Map<String, Serializable> properties,
+            Map<String, Set<String>> policy) {
+        HashMap<String, Set<String>> operationPolicy = new HashMap<>();
+        if (Requests.isLocal(properties) && input != null && (
+                RegistryUtility.isInternalRegistryMetacard(input)
+                        || RegistryUtility.isRegistryMetacard(input))) {
             String registryBaseUrl = RegistryUtility.getStringAttribute(input,
                     RegistryObjectMetacardType.REGISTRY_BASE_URL,
                     null);
             if (isRegistryDisabled() || (registryBaseUrl != null && registryBaseUrl.startsWith(
                     SystemBaseUrl.getBaseUrl()))) {
-                policy.putAll(bypassAccessPolicy);
+                operationPolicy.putAll(bypassAccessPolicy);
             } else {
-                policy.putAll(writeAccessPolicy);
+                operationPolicy.putAll(policy);
             }
         }
-        return new PolicyResponseImpl(policy, policy);
+        return new PolicyResponseImpl(operationPolicy, operationPolicy);
     }
 
     @Override
     public PolicyResponse processPreCreate(Metacard input, Map<String, Serializable> properties)
             throws StopProcessingException {
-        return getWritePolicy(input, properties);
+        return getWritePolicy(input, properties, createAccessPolicy);
     }
 
     @Override
     public PolicyResponse processPreUpdate(Metacard input, Map<String, Serializable> properties)
             throws StopProcessingException {
-        return getWritePolicy(input, properties);
+        return getWritePolicy(input, properties, updateAccessPolicy);
     }
 
     @Override
     public PolicyResponse processPreDelete(List<Metacard> metacards,
             Map<String, Serializable> properties) throws StopProcessingException {
         if (metacards != null) {
             for (Metacard metacard : metacards) {
-                PolicyResponse response = getWritePolicy(metacard, properties);
```

Comment:
deleteAccessPolicy is declared as an instance variable, why do you need to pass it in as an argument?